tiledb/sm/c_api/tiledb_filestore.cc

```diff
@@ -388,40 +472,56 @@ TILEDB_EXPORT int32_t tiledb_filestore_uri_export(
   uint64_t end_range = std::min(file_size, *buffer_size) - 1;
   do {
     uint64_t write_size = end_range - start_range + 1;
-    Subarray subarray(context, array);
-    subarray.add_range(0, start_range, end_range);
-    Query query(context, array);
-    query.set_layout(TILEDB_ROW_MAJOR);
+    tiledb::sm::Subarray subarray(
+        array.get(),
+        nullptr,
+        context.storage_manager()->logger(),
+        true,
+        context.storage_manager());
+    uint64_t subarray_range_arr[] = {start_range, end_range};
+    tiledb::type::Range subarray_range(
+        static_cast<void*>(subarray_range_arr), sizeof(uint64_t) * 2);
+    throw_if_not_ok(subarray.add_range(0, std::move(subarray_range)));
+
+    tiledb::sm::Query query(context.storage_manager(), array);
+    throw_if_not_ok(query.set_layout(tiledb::sm::Layout::ROW_MAJOR));
     query.set_subarray(subarray);
 
     // Cloud compatibility hack. Currently stored tiledb file arrays have a
     // TILEDB_UINT8 attribute. We should pass the right datatype here to
     // support reads from existing tiledb file arrays.
+    auto&& [st, schema] = array->get_array_schema();
```

Comment:
preserve the comment block position

Reply:
Hmmm, I think it is preserved? I just have to get the schema before I extract the attribute type, so it seems like it is not.

Reply:
My bad, you're correct